r/R/dplyr-functions.R

```diff
@@ -694,6 +694,10 @@ nse_funcs$wday <- function(x, label = FALSE, abbr = TRUE, week_start = getOption
 }
 
 nse_funcs$log <- nse_funcs$logb <- function(x, base = exp(1)) {
+  if (inherits(base, "Expression")) {
+    return(Expression$create("logb_checked", x, base))
+  }
+
```

Comment:
Should we assert anything about `base` here? Length 1, numeric? (If length(base) > 1, the we'll get more warnings about `if()` getting more than one condition.)

Reply:
inherits() is fine, we just need the check after it. You're right that length > 1 works in R but it won't here because we don't know the correct length that the supplied vector should be at this point, and more importantly, row order in Arrow queries isn't deterministic.